### PR TITLE
fix: prevent native credentials window

### PR DESCRIPTION
**Pull request by churkin**
\cc @miherlosev @MarinaRukavitsyna

### CODE REVIEW

File: test/server/auth-test.js [hunk 1]
```diff
@@ -126,7 +126,8 @@ describe('Authentication', function () {
             request(proxy.openSession('http://127.0.0.1:1507/', session), function (err, res, body) {
                 expect(body).equal('Access denied');
                 expect(res.statusCode).equal(401);
-
+                // NOTE: prevent native browser credentials window.
```
Comment (MarinaRukavitsyna) [inline test/server/auth-test.js]: // prevent showing the native credentials window

File: src/request-pipeline/header-transforms.js [hunk 2]
```diff
@@ -92,6 +92,9 @@ var responseTransforms = {
     'x-content-security-policy-report-only': skip,
     'x-webkit-csp':                          skip,
 
+    // NOTE: Even if we were not able to authorize we should prevent native browser credentials window.
```
Comment (MarinaRukavitsyna) [inline src/request-pipeline/header-transforms.js]: // NOTE: Even if we are not able to authorize a user, we should prevent showing the native credentials window.

Comment (MarinaRukavitsyna) [inline src/request-pipeline/header-transforms.js]: // NOTE: Even if we are not able to be authorized, we should prevent showing the native credentials window.

**Comment by jshor** [comment]
The comments on this are useless and there is no explanation on the PR as to what this even does. What is a "native credentials window"? Why should it not be shown? Why was this merged?

**Pull request merged**